atomic_reactor/utils/cachi2.py

```diff
@@ -68,3 +68,49 @@ def convert_SBOM_to_ICM(sbom: Dict[str, Any]) -> Dict[str, Any]:
         {"purl": comp["purl"]} for comp in sbom["components"]  # type: ignore
     ]
     return icm
+
+
+def generate_request_json(
+    remote_source: Dict[str, Any], remote_source_sbom: Dict[str, Any],
+    remote_source_env_json: Dict[str, Dict[str, str]],
+) -> Dict[str, Any]:
+    """Generates Cachito like request.json
+
+    Cachito does provide request.json, for backward compatibility
+    as some tools are depending on it, we have to generate also request.json from cachi2
+    """
+    def gen_dependency(sbom_dep: Dict[str, Any]) -> Dict[str, str]:
+        """Generate a single dependency from SBOM"""
+        # we need to detect type from purl, this is just heuristics,
+        # we cannot reliably construct type from purl
+        heuristic_type = "unknown"
+        purl_type_map = {
+            "pkg:golang/": "go-package",
+            "pkg:npm/": "npm",
+            "pkg:pypi/": "pip",
+            "pkg:rpm/": "rpm",
+            "pkg:gem/": "rubygems",
+        }
+        purl = sbom_dep["purl"]
+        for purl_prefix, request_type in purl_type_map.items():
+            if purl.startswith(purl_prefix):
+                heuristic_type = request_type
+                break
+
+        return {
+            "name": sbom_dep["name"],
+            # "replaces": None, #  TBD: this is go only, but will be always none as we don't support replacements in cachi2; should it be included?
```

Comment:
It is optional per the API spec https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/web/static/api_v1.yaml#L897-L900

But I think Cachito always included it, so we can keep it to be safe

Reply:
updated